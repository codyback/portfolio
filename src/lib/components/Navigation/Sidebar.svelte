<script lang="ts">
  import { fly } from 'svelte/transition';

  function toggleDrawer() {
    open = !open;
  }

  const links = [
    {
      href: '/about',
      title: 'About Page',
      label: 'About',
    },
    // {
    //   href: '/projects',
    //   title: 'Projects Page',
    //   label: 'Projects',
    // },
    // {
    //   href: '/blog',
    //   title: 'Blog Page',
    //   label: 'Blog',
    // },
    // {
    //   href: '/resources',
    //   title: 'Resources Page',
    //   label: 'Resources',
    // },
  ];

  export let open = false;
</script>

{#if open}
  <aside
    transition:fly={{ x: 640, opacity: 1 }}
    class="absolute w-full h-full z-10 bg-gray-200 dark:bg-dark-700 dark:text-neutral-300 dark:border-none border-l-2 shadow-lg"
    class:open
  >
    <nav class="p-4 mt-14 text-xl">
      {#each links as link}
        <a
          href={link.href}
          class="px-4 py-2 block rounded-md hover:bg-slate-200 dark:hover:bg-dark-900 leading-10 uppercase"
          title={link.title}
          aria-label={link.title}
          on:click={toggleDrawer}
        >
          {link.label}
        </a>
      {/each}
    </nav>
  </aside>
{/if}

<style>
  aside {
    position: fixed;
    top: 0;
    right: 0;
    transition: right 0.3s ease-in-out;
  }

  .open {
    right: 0;
  }
</style>
