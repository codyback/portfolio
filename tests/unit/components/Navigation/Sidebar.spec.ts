import Sidebar from '@/lib/components/Navigation/Sidebar.svelte';
import { render } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';

describe('Sidebar', () => {
  it('renders correctly with open false', () => {
    const { container } = render(Sidebar, { open: false });

    expect(container).toMatchSnapshot();
  });

  it('renders correctly with open true', () => {
    const { container } = render(Sidebar, { open: true });

    expect(container).toMatchSnapshot();
  });
});
