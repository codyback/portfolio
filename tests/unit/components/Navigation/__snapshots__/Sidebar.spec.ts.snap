// Vitest Snapshot v1

exports[`Sidebar > renders correctly with open false 1`] = `
<body>
  <div>
    
    <!--&lt;Sidebar&gt;-->
  </div>
</body>
`;

exports[`Sidebar > renders correctly with open true 1`] = `
<body>
  <div>
    <aside
      class="absolute w-full h-full z-10 bg-gray-200 dark:bg-dark-700 dark:text-neutral-300 dark:border-none border-l-2 shadow-lg s-RQ-jkbaw6IIv open"
    >
      <nav
        class="p-4 mt-14 text-xl s-RQ-jkbaw6IIv"
      >
        <a
          aria-label="About Page"
          class="px-4 py-2 block rounded-md hover:bg-slate-200 dark:hover:bg-dark-900 leading-10 uppercase s-RQ-jkbaw6IIv"
          href="/about"
          title="About Page"
        >
          About
           
        </a>
        <a
          aria-label="Projects Page"
          class="px-4 py-2 block rounded-md hover:bg-slate-200 dark:hover:bg-dark-900 leading-10 uppercase s-RQ-jkbaw6IIv"
          href="/projects"
          title="Projects Page"
        >
          Projects
           
        </a>
        <a
          aria-label="Blog Page"
          class="px-4 py-2 block rounded-md hover:bg-slate-200 dark:hover:bg-dark-900 leading-10 uppercase s-RQ-jkbaw6IIv"
          href="/blog"
          title="Blog Page"
        >
          Blog
           
        </a>
        <a
          aria-label="Resources Page"
          class="px-4 py-2 block rounded-md hover:bg-slate-200 dark:hover:bg-dark-900 leading-10 uppercase s-RQ-jkbaw6IIv"
          href="/resources"
          title="Resources Page"
        >
          Resources
           
        </a>
      </nav>
    </aside>
    
    <!--&lt;Sidebar&gt;-->
  </div>
</body>
`;
